### PR TITLE
Added UI Animations for GPS Loading | Store/Item Placeholders waiting for server response

### PR DESCRIPTION
**Pull request by devYaoYH**
- Added css animations in `src/css/animation.css`
- Introduced `MediaInfoCard` component for common card layout for both `ItemCard` and `StoreCard`
- Introduced `InfoCardMedia` sub-component to handle image media height adjustments logic
- Introduced `isLoading` flag for `IconSearchBar` to handle adding 'loading' animation when triggering action via icon click (GPS loading)
- Added `StorePlaceholder` and `ItemPlaceholder` based on common `MediaInfoCard` layout
  - Used `<div />` with styled height and width to mock actual text regions
  - Added `idle-gradient` class to trigger 'gradient wipe' animation to div background
  - Placeholders pass a `placholder` boolean flag into `MediaInfoCard` which forwards it to `InfoCardMedia` as well to add the required classes for css animations
- Removed obsolete sub-components `StoreCardMedia` and `ItemCardMedia` now that image rendering logic accompanying Store or Item display is handled by `MediaInfoCard` via `InfoCardMedia` sub-component
- Updated `StoreList` and `Cart` to take `null` values for array of stores or items respectively and render Placeholders instead of typical `StoreCard` or `ItemCard` components
- Updated `Home` and `ScanStore` pages to pass/add `null` values into arrays passed to `StoreList`/`Cart` to render placeholders while waiting for GPS/Server callback
- Introduced `themeConfig` exported object from `src/theme.ts` to replace usage of `MAX_CARD_MEDIA_HEIGHT`-ish variables declared in `StoreCard` and `ItemCard` so far. Ensures our UI-related configuration settings are in one common place and used consistently throughout app (like `useTheme()` and `MuiThemeProvider` pattern also declared in `src/theme.ts`)
- **Bug** Fixed issue where `InfoCardMedia` was not waiting for image to load to scale height and width to fit parent div

Demo:
ScanAndGo GPay (2)

### CODE REVIEW

File: client/src/utils.ts [hunk 1]
```diff
@@ -194,3 +194,8 @@ export const jestImportImage = (imgPath: string) => {
   // Format this information into src expected format
   return `data:image/${imgFileType};base64,${imgFileData}`;
 };
+
+// Fill an empty array
+export const emptyArray = (length: number) => {
+  return new Array(length).fill(null);
```
Comment (BhavyaLight) [inline client/src/utils.ts:200]: i meant here -> use immutables

File: client/src/components/Cart/Cart.tsx [hunk 2]
```diff
@@ -10,20 +10,24 @@ import {
...
 import ItemCardCompact from "src/components/ItemCardCompact";
+import ItemPlaceholder from "src/components/ItemPlaceholder";
 
 function Cart({
   contents,
   collapse,
   updateItemQuantity,
 }: {
-  contents: CartItem[];
+  contents: (CartItem | null)[];
```
Comment (BhavyaLight) [inline client/src/components/Cart/Cart.tsx:20]: avoid using nulls in code.

File: client/src/components/Cart/Cart.tsx [hunk 3]
```diff
@@ -10,20 +10,24 @@ import {
...
   updateItemQuantity?: (barcode: string, quantity: number) => void;
 }) {
   return (
     <div className="Cart">
       {contents.length > 0 &&
-        contents.map((cartItem) => {
+        contents.map((cartItem: CartItem | null, i: number) => {
+          if (!cartItem) {
+            return <ItemPlaceholder key={`item-${i}`} />;
+          }
```
Comment (BhavyaLight) [inline client/src/components/Cart/Cart.tsx:30]: I think the loading logic and rendering logic should be 2 different components. 
They should be integrated in this manner:

```
{isLoading ?
<ShowLoadingCart length={6} /> : <Cart ...props />
```

File: client/src/utils.ts (hunk 1 above)
Comment (BhavyaLight) [reply client/src/utils.ts:200]: avoid adding null/undefined values to your code

File: client/src/components/StoreList/StoreList.tsx [hunk 4]
```diff
@@ -20,13 +21,17 @@ function StoreList({ stores }: { stores: Store[] }) {
...
     <div className="StoreList">
       {stores &&
-        stores.map((store) => (
-          <StoreCard
-            key={store["store-id"]}
-            redirect={enterStore}
-            store={store}
-          />
-        ))}
+        stores.map((store: Store | null, i: number) =>
```
Comment (BhavyaLight) [inline client/src/components/StoreList/StoreList.tsx:24]: same here, avoid null and keep StoreList rendering and GhostLoading animation separate on a top level component.

File: client/src/components/Cart/Cart.tsx (hunk 3 above)
Comment (devYaoYH) [reply client/src/components/Cart/Cart.tsx]: adopted model of separate list components for placholder vs real cards

File: client/src/components/Cart/Cart.tsx (hunk 2 above)
Comment (devYaoYH) [reply client/src/components/Cart/Cart.tsx]: removed.

File: client/src/utils.ts (hunk 1 above)
Comment (devYaoYH) [reply client/src/utils.ts:200]: removed unused function

File: client/src/components/StoreList/StoreList.tsx (hunk 4 above)
Comment (devYaoYH) [reply client/src/components/StoreList/StoreList.tsx]: added ghostloading components `PlaceholderStoreList` and `PlaceholderCart`

**Pull request merged**